blocks/library/gallery/block.js

```diff
@@ -55,10 +56,27 @@ class GalleryBlock extends Component {
 	}
 
 	onSelectImage( index ) {
-		return () => {
+		return ( event ) => {
+			// ignore clicks in the editable caption.
+			// Without this logic, text operations like selection, select / unselects the images.
+			if ( event.target.tagName === 'FIGCAPTION' ) {
```

Comment:
What if deselecting a gallery item works similarly to selecting and deselecting blocks? So a click inside the block would not deselect the block, only a click outside it or in another block would.

Reply:
> What if deselecting a gallery item works similarly to selecting and deselecting blocks? So a click inside the block would not deselect the block, only a click outside it or in another block would.

That's a nice suggestion, and I think it may be a good idea, but maybe we can do it in a different PR so we can discuss this change and receive more feedback if this is something we should pursue or no.

Reply:
Well, I think this change would make the image and gallery block behave consistently and at the same time simplify the code.